src/checkers/inference/DefaultSlotManager.java

```diff
@@ -330,18 +329,18 @@ public ConstantSlot createConstantSlot(AnnotationMirror value) {
     }
 
     @Override
-    public CombVariableSlot createCombVariableSlot(Slot receiver, Slot declared) {
-        CombVariableSlot combVariableSlot;
+    public TernaryVariableSlot createTernaryVariableSlot(Slot receiver, Slot declared) {
+        TernaryVariableSlot ternaryVariableSlot;
         Pair<Slot, Slot> pair = new Pair<>(receiver, declared);
-        if (combSlotPairCache.containsKey(pair)) {
-            int id = combSlotPairCache.get(pair);
-            combVariableSlot = (CombVariableSlot) getVariable(id);
+        if (ternarySlotPairCache.containsKey(pair)) {
+            int id = ternarySlotPairCache.get(pair);
+            ternaryVariableSlot = (TernaryVariableSlot) getVariable(id);
         } else {
-            combVariableSlot = new CombVariableSlot(null, nextId(), receiver, declared);
-            addToVariables(combVariableSlot);
-            combSlotPairCache.put(pair, combVariableSlot.getId());
+            ternaryVariableSlot = new TernaryVariableSlot(null, nextId(), receiver, declared);
```

Comment:
`combVariableSlot` was used to store `VPResultSlot` before. But here, you used `TernaryVariableSlot`. Are you going to use `TernaryVariableSlot` to model viewpoint adaptation result or you want to generalize the API to add all kinds of `TernaryVariableSlot` , for example, `VPResultSlot`, `LeastUpperBoundSlot`? For the first case, it doesn't make sense for me. If the latter case is what you intend to do, why removing API to add `VPResultSlot`?